Std/Tactic/ByCases.lean

```diff
@@ -74,10 +75,16 @@ syntax (name := tacIfThenElse)
   ppRealGroup(ppRealFill(ppIndent("if " term " then") ppSpace matchRhs)
     ppDedent(ppSpace) ppRealFill("else " matchRhs)) : tactic
 
-macro_rules
+elab_rules : tactic
   | `(tactic| if%$tk $h : $c then%$ttk $pos else%$etk $neg) =>
     expandIfThenElse tk ttk etk pos neg fun pos neg => `(if $h : $c then $pos else $neg)
 
-macro_rules
+elab_rules : tactic
   | `(tactic| if%$tk $c then%$ttk $pos else%$etk $neg) =>
     expandIfThenElse tk ttk etk pos neg fun pos neg => `(if h : $c then $pos else $neg)
+
+-- This is an improved version of `by_cases` from core that uses `Decidable` if possible
+macro_rules
+  | `(tactic| by_cases $h : $e) =>
+    `(tactic| open Classical in if $h:ident : $e then _ else _)
+macro_rules | `(tactic| by_cases $e) => `(tactic| by_cases h : $e)
```

Comment:
It seems like this is an orthogonal change, which should be in a separate PR.

Reply:
No, it was moved from above and made the changes necessary for this PR.

Reply:
Oops, thanks.